assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -199,26 +217,99 @@ const SelectionCardContainer = ({
 }: {
   children: JSX.Element
 }): JSX.Element => {
+  // #region Catching Events Before Leaflet
+  const cancelEvent: MouseEventHandler<HTMLDivElement> = (e) => {
+    e.stopPropagation()
+  }
+
+  const keepUserInputFromLeaflet: ComponentPropsWithoutRef<"div"> = {
+    onMouseDownCapture: cancelEvent,
+    onDoubleClickCapture: cancelEvent,
+    onScrollCapture: cancelEvent,
+    onWheelCapture: cancelEvent,
+  }
+  // #endregion
   return (
-    <div className="m-map-display__selected-entity-card-container">
+    <div
+      {...keepUserInputFromLeaflet}
+      className="m-map-display__selected-entity-card-container"
+    >
       {children}
     </div>
   )
 }
 
+const RoutePatternLayers = ({
+  routePattern,
+  isSelected,
+  onShapeClick,
+}: {
+  routePattern: RoutePattern
+  isSelected: boolean
+  onShapeClick: () => void
+}): JSX.Element => {
+  return routePattern.shape ? (
+    <>
+      <ZoomLevelWrapper>
+        {(zoomLevel) => {
+          const routePatternIdSplit = routePattern.id.split("-")
+          const variantFormatted =
+            routePatternIdSplit.length === 3 && routePatternIdSplit[1] !== "_"
+              ? "_" + routePatternIdSplit[1]
+              : ""
+          return (
+            <>
+              {routePattern.shape && (
+                <>
+                  <RouteShape
+                    shape={routePattern.shape}
+                    isSelected={isSelected}
+                    onClick={onShapeClick}
+                  >
+                    {zoomLevel >= 16 && !isSelected ? (
+                      <Tooltip
+                        className="route-shape__tooltip"
+                        sticky={true}
+                        direction="top"
+                      >
+                        Click to select route {routePattern.routeId}
+                        {variantFormatted}.
+                      </Tooltip>
+                    ) : undefined}
```

Comment:
suggestion: instead of the ternary operator, I think you could do this with the `&&` operator?

Reply:
It doesn't like the type Element | false, but `&& !isSelected ... || undefined` does the trick!